src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -183,6 +247,21 @@ public void run(SourceContext<T> ctx) throws Exception {
 
             log.info("Starting Pravega reader '{}' for controller URI {}", readerId, this.clientConfig.getControllerURI());
 
+            long previousTimestamp = Long.MIN_VALUE;
+            AssignerWithTimeWindows<T> assigner = null;
+            // If it is event time, register a watermark emitter
+            if (isEventTimeMode()) {
+                assigner = assignerWithTimeWindows.deserializeValue(getRuntimeContext().getUserCodeClassLoader());
+                PeriodicWatermarkEmitter periodicEmitter = new PeriodicWatermarkEmitter(
+                        pravegaReader,
+                        ctx,
+                        getRuntimeContext().getUserCodeClassLoader(),
+                        ((StreamingRuntimeContext) getRuntimeContext()).getProcessingTimeService(),
+                        getRuntimeContext().getExecutionConfig().getAutoWatermarkInterval());
```

Comment:
It is `0`: https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java#L125, I'll add it as an variable of the class to make it easier to use and check. I'd like to start another conversation here about this value. Since Pravega now updates the `TimeWindows` every 10 seconds, should we add more checks like ` interval > 10000ms` because any duration much shorter that 10 seconds is now actually useless as `TimeWindow` keeps the same.

Reply:
It looks like it gets changed to `200` as side effect of changing the Time Characteristic.

https://github.com/apache/flink/blob/7ea00ad9004f920e0c4c1b6ed906e14418951272/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#L642-L649 but obviously we should still have some validation as a user can change this AFTER setting the Time Characteristic 

```
	public void setStreamTimeCharacteristic(TimeCharacteristic characteristic) {
		this.timeCharacteristic = Preconditions.checkNotNull(characteristic);
		if (characteristic == TimeCharacteristic.ProcessingTime) {
			getConfig().setAutoWatermarkInterval(0);
		} else {
			getConfig().setAutoWatermarkInterval(200);
		}
	}
```

Reply:
Thanks for correcting me. I added a precondition check during the `FlinkPravegaReader` `open` function. That's the first place where the `RuntimeContext` is being initialized and accessible.